contracts/multisigs/GuardCM.sol

```diff
@@ -0,0 +1,292 @@
+    Expired,
+    Executed
+}
+
+/// @dev Only `owner` has a privilege, but the `sender` was provided.
+/// @param sender Sender address.
+/// @param owner Required sender address as an owner.
+error OwnerOnly(address sender, address owner);
+
+/// @dev Only `manager` has a privilege, but the `sender` was provided.
+/// @param sender Sender address.
+/// @param manager Required sender address as an owner.
+error ManagerOnly(address sender, address manager);
+
+/// @dev Provided zero address.
+error ZeroAddress();
+
+/// @dev Provided zero value.
+error ZeroValue();
+
+/// @dev Wrong length of two arrays.
+/// @param numValues1 Number of values in a first array.
+/// @param numValues2 Numberf of values in a second array.
+error WrongArrayLength(uint256 numValues1, uint256 numValues2);
+
+/// @dev Provided incorrect data length.
+/// @param expected Expected minimum data length.
+/// @param provided Provided data length.
+error IncorrectDataLength(uint256 expected, uint256 provided);
+
+/// @dev No delegatecall is allowed.
+error NoDelegateCall();
+
+/// @dev No self multisig call is allowed.
+error NoSelfCall();
+
+/// @dev The combination of target and selector is not authorized.
+/// @param target Target address.
+/// @param selector Function selector.
+error NotAuthorized(address target, bytes4 selector);
+
+/// @dev The proposal is not defeated.
+/// @param proposalId Proposal Id.
+/// @param state Current proposal state.
+error NotDefeated(uint256 proposalId, ProposalState state);
+
+/// @title GuardCM - Smart contract for Gnosis Safe community multisig (CM) guard
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+contract GuardCM {
+    event GovernorUpdated(address indexed governor);
+    event SetTargetSelectors(address[] indexed targets, bytes4[] indexed selectors, bool[] statuses);
+    event GovernorCheckProposalIdChanged(uint256 indexed proposalId);
+    event GuardPaused(address indexed account);
+    event GuardUnpaused();
+
+    // Owner address
+    address public immutable owner;
+    // Multisig address
+    address public immutable multisig;
+    // Governor address
+    address public governor;
+    // Guard pausing possibility
+    uint8 public paused = 1;
+    
+    // Mapping of address + bytes4 selector => enabled / disabled
+    mapping(uint256 => bool) public mapAllowedTargetSelectors;
+
+    // Schedule selector
+    bytes4 public constant SCHEDULE = bytes4(keccak256(bytes("schedule(address,uint256,bytes,bytes32,bytes32,uint256)")));
+    // ScheduleBatch selector
+    bytes4 public constant SCHEDULE_BATCH = bytes4(keccak256(bytes("scheduleBatch(address[],uint256[],bytes[],bytes32,bytes32,uint256)")));
+    // Initial check governance proposal Id
+    // Calculated from the proposalHash function of the GovernorOLAS with the following parameters:
+    // targets = [address(0)], values = [0], calldatas = [0x], description = "Is governance alive?"
```

Comment:
We cannot have more than one governance proposal with all the same data.
So we should have
// targets = [address(0)], values = [0], calldatas = [0x], description = "Is governance alive - test number x?" ,
where x increases with the increase of this kind of proposal

Reply:
"Is governance alive?"
vs
"Is governance alive??"
vs
"Is governance alive - test number x?"
since it's a hash of a human-readable string, the text of the pattern itself doesn't matter.
You just need to agree on what text will be in next "check alive" proposalId. And change it. See above.
descriptionHash (bytes32 which itself is the keccak256 hash of the description string)
uint256 proposalId = hashProposal(targets, values, calldatas, descriptionHash);

Reply:
yeap. Agree, I was proposing a standard format we can use for this :-)

Reply:
Yeah, we can modify that at any time once the first proposal is used.